spec/index.md

```diff
@@ -6242,16 +6243,18 @@ Finally we can specify the abstract `CanisterModule` that models a concrete WebA
 
 -   The partial map `query_methods` of the `CanisterModule` is defined for all method names `method` for which the WebAssembly program exports a function `func` named `canister_query <method>`, and has value
 
-        query_methods[method] = λ (arg, caller, sysenv) → λ wasm_state →
+        query_methods[method] = λ (arg, caller, sysenv, available) → λ wasm_state →
           let es = ref {empty_execution_state with
               wasm_state = wasm_state;
               params = empty_params with { arg = arg; caller = caller; sysenv }
               balance = sysenv.balance
+              cycles_available = available;
               context = Q
             }
           try func<es>() with Trap then Trap {cycles_used = es.cycles_used;}
           Return {
             response = es.response;
+            cycles_accepted = es.cycles_accepted;
```

Comment:
This field should be read in the Message execution transition:
```
query_as_update(f, arg, env) = λ wasm_state →
  match f(arg, env)(wasm_state) with
    Trap trap → Trap trap
    Return res → Return {
      new_state = wasm_state;
      new_calls = [];
      new_certified_data = NoCertifiedData;
      new_global_timer = NoGlobalTimer;
      response = res.response;
      cycles_accepted = res.cycles_accepted;
      cycles_used = res.cycles_used;
    }
```

Reply:
Done.